### PR TITLE
Add version check to reported output

### PR DESCRIPTION
**Pull request by mosuem**
## Description
For our [health.yaml](https://github.com/dart-lang/ecosystem/blob/main/.github/workflows/health.yaml) workflow, it would be ideal if the version checking would also be part of the JSON report.

## Type of Change

- [x] 🚀 New feature (non-breaking change)
- [ ] 🛠️ Bug fix (non-breaking change)
- [x] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [ ] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [ ] 📝 Documentation
- [ ] 🧹 Chore / Housekeeping

### CODE REVIEW

File: lib/src/cli/commands/version_check.dart [hunk 1]
```diff
@@ -5,19 +5,40 @@ import 'package:pub_semver/pub_semver.dart';
...
-  static bool versionChangeMatchesChanges({
+  static VersionCheckResult check({
     required PackageApiDiffResult diffResult,
     required PackageApi oldPackageApi,
     required PackageApi newPackageApi,
     required bool ignorePrerelease,
     required VersionCheckMode versionCheckMode,
   }) {
-    stdout.writeln('');
     stdout.writeln('Checking Package version');
```
Comment (mosuem) [inline lib/src/cli/commands/version_check.dart:41]: I don't know what you want to still `stdout` here.

File: lib/src/cli/commands/version_check.dart [hunk 2]
```diff
@@ -57,28 +85,36 @@ abstract class VersionCheck {
...
+          'We got a pre release. We only check if there are any changes.';
       if (containsAnyChanges && oldVersion >= newVersion) {
-        stdout.writeln(
-            'Got "${Colorize(newVersion.toString()).bold()}" expected > "${Colorize(oldVersion.toString()).bold()}" (pre-release but changes)');
-        return false;
+        return VersionCheckResult.failure(
+          oldVersion: oldVersion,
+          newVersion: newVersion,
+          neededVersion: null,
+          explanation:
```
Comment (mosuem) [inline lib/src/cli/commands/version_check.dart:106]: Happy to change the explanations to anything you like better.

File: test/unit_tests/cli/commands/version_check_test.dart [hunk 3]
```diff
@@ -40,126 +40,126 @@ void main() {
 
   group('Version Check', () {
     test('is fine with breaking change and major version change', () {
-      bool versionChangeCheckResult = VersionCheck.versionChangeMatchesChanges(
+      final versionChangeCheckResult = VersionCheck.check(
```
Comment (mosuem) [inline test/unit_tests/cli/commands/version_check_test.dart:43]: The tests could be more fine-grained now probably, also checking for expected version etc.

**Comment by devmil** [comment]
Thanks for your changes! 
Did you try what happens if your generate a JSON or Markdown report in regards to the Colorize usage? I assume that the terminal control sequences would end up there, right?
If that's the case I would suggest to get rid of those in the source and maybe add a couple of them back to the Console Reporter (where it still is possible to grab the affected text pieces)

**Comment by devmil** [comment]
I allowed myself to adapt your PR a bit 😀
Thanks for your contribution! Will release a new version as soon as I merged this PR

**Review by devmil — APPROVED** [review]

**Comment by mosuem** [comment]
> I allowed myself to adapt your PR a bit 😀 Thanks for your contribution! Will release a new version as soon as I merged this PR

Thanks a lot! I was OOO :)

**Pull request merged**